l10n_ch_payment_slip/payment_slip.py

```diff
@@ -483,9 +483,6 @@ def _draw_address(self, canvas, print_settings, initial_position, font,
         text.moveCursor(0.0, font.size)
 
         address_lines = com_partner.contact_address.split("\n")
-        if com_partner.country_id:
```

Comment:
Here is a proposal for 9.0 https://github.com/OCA/l10n-switzerland/pull/303 and future versions
A bit more complicated but should cover all the needs and by default will work without further configuration.